.github/workflows/README.md

```diff
@@ -0,0 +1,32 @@
+# Use workflows in forked repositories
+
+`forked-helper.yml` workflow helper can help to run custom workflows on the forked repositories.
```

Comment:
Just to make sure I understand this right, the work flow is dispatching events to a specific branch, so that a fork can maintain its own workflows on that branch, correct?

Can you maybe extend that description a bit?

Reply:
Repository dispatch event always processed [by the default branch](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#repository_dispatch) of the github repository.

So, suggested helper will forward events on **any** branch to the default branch.

Forked repository owner should create **empty** default branch just with workflows and some optional files like `README.md`, etc.

Reply:
Why not just have a custom workflow in your forked branch? Adding a workflow is unlikely to ever cause conflicts, and then you can maintain work flows for each branch. E.g. for Home Assistant OS, we add and remove machines, so you want to have a different work flow configuration for `rel-6` compared to `rel-7`. Your rel-6 branch will be slightly different to upstream `rel-6`, but it will be clean merge commits in the future.

As for the repository dispatch event, as you write, these anyway get processed on the default branch. So you can just define a different default branch in your fork.

Reply:
This workflow in your repository will dispatch events to forked repos. So forks can do something based on events (push, pr etc) from your repo. This will make life easier for forks.

Reply:
> This workflow in your repository will dispatch events to forked repos. So forks can do something based on events (push, pr etc) from your repo. This will make life easier for forks.


No. Its dispath event only inside fork.

Main goal is to keep fork clear without modifications of the `.github/workflows` folder.

Of cause, we can put any custom workflow inside forked branch, but in this case this workflow changes **will appear in PR** which will be made from this branch.

So, with this helper fork owner can store all custom worflows **outside of the forked branch**, in custom empty branch.

@agners - is it clear for you now?